ckan/lib/helpers.py

```diff
@@ -868,7 +868,11 @@ def dict_list_reduce(list_, key, unique=True):
     values for the key with unique values if requested. '''
     new_list = []
     for item in list_:
-        value = item.get(key)
+        try:
+            value = item.get(key)
+        except AttributeError:
+            # item might not be a dict.
+            continue
```

Comment:
I think the thing to investigate is why when calling ```dict_list_reduce``` with 

```h.dict_list_reduce(package.resources, 'format')```

we get things that are not resource dicts.  package.resources should *definitely* be a list of dicts.

why one of the items is an int and not a dict.

Perhaps just dumping ```{{package.resources}}``` in the package_item template might show what it is being passed, I can only imagine that a plugin/custom-schema is interfering somehow.